src/_pytest/python.py

```diff
@@ -1586,7 +1588,10 @@ def __init__(
         # to a readonly property that returns FunctionDefinition.name.
 
         self.keywords.update(self.obj.__dict__)
-        self.own_markers.extend(get_unpacked_marks(self.obj))
+        self.own_markers.extend(get_marks_as_list(self.obj))
+        if self.cls:
+            extract_mro_markers(self.cls)
+            self.own_markers.extend(getattr(self.cls, "mro_markers"))
```

Comment:
all  mro markers should go the `Class` instances "own_markers"

Reply:
Yes, you're right, but it's a very big behavior change. Don't you want to do it in two parts?

Reply:
i dont follow, instead of the `if self.cls`, the logic just goes to the `Class` itself to set own_markers

basically for class, it would use the mro markers instead of the mark_as_list

the markers of a class should never materealize on a function